File: LittleProjects/Java/SpringHibernateCRUD/src/main/java/core/controller/AuthorController.java
package core.controller;

import core.model.Author;
import core.service.AuthorService;
import org.apache.log4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping(value = "/author")
public class AuthorController {
    private static Logger logger = Logger.getLogger(AuthorController.class.getSimpleName());
    private final AuthorService service;

    @Autowired
    public AuthorController(AuthorService service) {
        this.service = service;
    }

    @GetMapping(value = "/select/{id}", produces = "application/json")
    public Author authorById(@PathVariable Long id) {
        return service.select(id);
    }

    @GetMapping(value = "/select", produces = "application/json")
    public Iterable<Author> allAuthors() {
        return service.select();
    }

    @DeleteMapping(value = "/delete/{id}")
    public void deleteAuthorById(@PathVariable Long id) {
        service.deleteById(id);
    }

    @DeleteMapping(value = "/delete")
    public void deleteAuthorsByIds(@RequestBody Iterable<Long> ids) {
        service.deleteById(ids);
    }

    @PutMapping(value = "/save", produces = "application/json")
    public Author saveAuthor(@RequestBody Author author) {
        return service.insert(author);
    }

    @PutMapping(value = "/save/all", produces = "application/json")
    public Iterable<Author> saveAuthors(@RequestBody Iterable<Author> authors) {
        return service.insert(authors);
    }

    @PostMapping(value = "/update", produces = "application/json")
    public Author updateAuthor(@RequestBody Author author) {
        return service.update(author);
    }

    @PostMapping(value = "/update/all", produces = "application/json")
    public Iterable<Author> updateAuthors(@RequestBody Iterable<Author> authors) {
        return service.update(authors);
    }
}

File: LittleProjects/Scala/ScalaSpringSimpleCRUD/src/main/java/com/github/gr1f0n6x/controller/AuthorController.scala
package com.github.gr1f0n6x.controller

import com.github.gr1f0n6x.service.AuthorService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation._

@RestController
@RequestMapping(value = Array("/author"))
class AuthorController {
  @Autowired
  var service: AuthorService = _

  @GetMapping(value = Array("/{id}"))
  def get(@PathVariable id: Long): String = service.select(id)

  @GetMapping(value = Array(""))
  def getAll(): String = service.select()

  @PutMapping(value = Array(""))
  def save(@RequestBody body: String): String = service.insert(body)

  @PostMapping(value = Array(""))
  def update(@RequestBody body: String): String = service.update(body)

  @DeleteMapping(value = Array("/{id}"))
  def delete(@PathVariable id: Long): String = service.delete(id)
}
